### PR TITLE
Remove tf.test.TestCase in beam search ops

### PR DESCRIPTION
**Pull request by autoih**
cc @gabrieldemarmiesse

### CODE REVIEW

File: tensorflow_addons/seq2seq/tests/beam_search_ops_test.py [hunk 1]
```diff
@@ -46,55 +47,52 @@ def test_gather_tree_one():
...
-        expected_result = _transpose_batch_time(
-            [[[2, -1, 2], [6, 5, 6], [7, 8, 9], [10, 10, 10]]]
-        )
-        with tf.device("/device:GPU:0"):
-            beams = gather_tree(
+def test_bad_parent_values_on_cpu():
+    # (batch_size = 1, max_time = 4, beams = 3)
+    # bad parent in beam 1 time 1
+    end_token = 10
+    step_ids = _transpose_batch_time([[[1, 2, 3], [4, 5, 6], [7, 8, 9], [-1, -1, -1]]])
```
Comment (qlzh727) [inline tensorflow_addons/seq2seq/tests/beam_search_ops_test.py:54]: Please wrap the long lines here.

File: tensorflow_addons/seq2seq/tests/beam_search_ops_test.py [hunk 2]
```diff
@@ -46,55 +47,52 @@ def test_gather_tree_one():
...
+
+def test_bad_parent_values_on_gpu():
+    # Only want to run this test on CUDA devices, as gather_tree is not
+    # registered for SYCL devices.
+    if not tf.test.is_gpu_available(cuda_only=True):
+        return
+    # (max_time = 4, batch_size = 1, beams = 3)
+    # bad parent in beam 1 time 1; appears as a negative index at time 0
+    end_token = 10
+    step_ids = _transpose_batch_time([[[1, 2, 3], [4, 5, 6], [7, 8, 9], [-1, -1, -1]]])
```
Comment (qlzh727) [inline tensorflow_addons/seq2seq/tests/beam_search_ops_test.py:80]: same.

File: tensorflow_addons/seq2seq/tests/beam_search_ops_test.py (hunk 1 above)
Comment (autoih) [reply tensorflow_addons/seq2seq/tests/beam_search_ops_test.py]: Thanks @qlzh727. I've updated it with pytest decorator. May need your review and comments.

File: tensorflow_addons/seq2seq/tests/beam_search_ops_test.py [hunk 3]
```diff
@@ -46,55 +47,27 @@ def test_gather_tree_one():
...
+    parent_ids = _transpose_batch_time(
+        [[[0, 0, 0], [0, -1, 1], [2, 1, 2], [-1, -1, -1]]]
+    )
+    max_sequence_lengths = [3]
+    expected_result = _transpose_batch_time(
+        [[[2, -1, 2], [6, 5, 6], [7, 8, 9], [10, 10, 10]]]
+    )
+    with pytest.raises(tf.errors.InvalidArgumentError):
```
Comment (gabrieldemarmiesse) [inline tensorflow_addons/seq2seq/tests/beam_search_ops_test.py:63]: I don't think it's possible to combine the cpu and gpu tests here. In the cpu tests, and error is thrown, in the gpu tests, everything works. So I believe that we should keep the tests separated.

**Comment by gabrieldemarmiesse** [comment]
Looks good to me be we need to wait for #1677 to be merged to make sure that it actually works on gpu.

**Comment by gabrieldemarmiesse** [comment]
@googlebot I consent.

**Pull request merged**